src/back/controllers/api/routes/challenge/challenge.controller.spec.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+const proxyquire = require('proxyquire').noCallThru(),
+    path = require('path'),
+    BaseController = require('../../../../common/base.controller'),
+    httpMocks = require('node-mocks-http'),
+    Q = require('q');
```

Comment:
We have sinon as promise plugged in, so you can do smth like `env.stub().resolve(1)()` to get a promise.